Source/VMware.vSphereDSC/DSCResources/VMHostPermission.ps1

```diff
@@ -0,0 +1,534 @@
+                }
+                else {
+                    $this.ModifyVMHostPermission($vmHostPermission)
+                }
+            }
+            else {
+                if ($null -ne $vmHostPermission) {
+                    $this.RemoveVMHostPermission($vmHostPermission)
+                }
+            }
+        }
+        finally {
+            $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.SetMethodEndMessage -Arguments @($this.DscResourceName)
+        }
+    }
+
+    [bool] Test() {
+        try {
+            Write-VerboseLog -Message $this.TestMethodStartMessage -Arguments @($this.DscResourceName)
+            $this.ConnectVIServer()
+            $this.EnsureConnectionIsESXi()
+
+            $this.RetrieveVMHost()
+
+            $foundEntityLocation = $this.GetEntityLocation()
+            $entity = $this.GetEntity($foundEntityLocation)
+            $vmHostPrincipal = $this.GetVMHostPrincipal()
+
+            $vmHostPermission = $this.GetVMHostPermission($entity, $vmHostPrincipal)
+            $result = $null
+
+            if ($this.Ensure -eq [Ensure]::Present) {
+                if ($null -eq $vmHostPermission) {
+                    $result = $false
+                }
+                else {
+                    $result = !$this.ShouldModifyVMHostPermission($vmHostPermission)
+                }
+            }
+            else {
+                $result = ($null -eq $vmHostPermission)
+            }
+
+            $this.WriteDscResourceState($result)
+
+            return $result
+        }
+        finally {
+            $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.TestMethodEndMessage -Arguments @($this.DscResourceName)
+        }
+    }
+
+    [VMHostPermission] Get() {
+        try {
+            Write-VerboseLog -Message $this.GetMethodStartMessage -Arguments @($this.DscResourceName)
+            $result = [VMHostPermission]::new()
+
+            $this.ConnectVIServer()
+            $this.EnsureConnectionIsESXi()
+
+            $this.RetrieveVMHost()
+
+            $foundEntityLocation = $this.GetEntityLocation()
+            $entity = $this.GetEntity($foundEntityLocation)
+            $vmHostPrincipal = $this.GetVMHostPrincipal()
+
+            $vmHostPermission = $this.GetVMHostPermission($entity, $vmHostPrincipal)
+            $this.PopulateResult($result, $vmHostPermission)
+
+            return $result
+        }
+        finally {
+            $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.GetMethodEndMessage -Arguments @($this.DscResourceName)
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Root Resource Pool from the VMHost.
+    #>
+    [PSObject] GetRootResourcePool() {
+        try {
+            $resourcePoolsType = (Get-ResourcePool -Server $this.Connection | Select-Object -First 1).GetType()
+            $rootResourcePool = Get-Inventory -Server $this.Connection -ErrorAction Stop -Verbose:$false |
```

Comment:
Check whether you can replace Get-Inventory with Get-ResourcePool

Reply:
Done.